application/src/main/java/org/togetherjava/tjbot/config/SuggestionsConfig.java

```diff
@@ -4,24 +4,26 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonRootName;
 
+import java.util.Objects;
+
 /**
  * Configuration for the suggestion system, see
  * {@link org.togetherjava.tjbot.commands.basic.SuggestionsUpDownVoter}.
  */
-@SuppressWarnings("ClassCanBeRecord")
 @JsonRootName("suggestions")
 public final class SuggestionsConfig {
     private final String channelPattern;
     private final String upVoteEmoteName;
     private final String downVoteEmoteName;
 
     @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
-    private SuggestionsConfig(@JsonProperty("channelPattern") String channelPattern,
-            @JsonProperty("upVoteEmoteName") String upVoteEmoteName,
-            @JsonProperty("downVoteEmoteName") String downVoteEmoteName) {
-        this.channelPattern = channelPattern;
-        this.upVoteEmoteName = upVoteEmoteName;
-        this.downVoteEmoteName = downVoteEmoteName;
+    private SuggestionsConfig(
+            @JsonProperty(value = "channelPattern", required = true) String channelPattern,
+            @JsonProperty(value = "upVoteEmoteName", required = true) String upVoteEmoteName,
+            @JsonProperty(value = "downVoteEmoteName", required = true) String downVoteEmoteName) {
```

Comment:
this is not required ig, we got fallback emote

https://github.com/Together-Java/TJ-Bot/blob/a5df7c5b20b37db1dd6510146966b92f1563a0e4/application/src/main/java/org/togetherjava/tjbot/commands/basic/SuggestionsUpDownVoter.java#L25-L26